### PR TITLE
feat(jshint): add initial config

### PR DESCRIPTION
**Pull request by denis-sokolov**
This initial config mostly documents existing code style
by relaxing multiple options that are turned on by default.

### CODE REVIEW

**Comment by paglias** [comment]
@denis-sokolov I'm not against enforcing some rules on code style but adding a .jshintrc alone would not change much. We should decide a-) if we want to enforce a certain code style b-) decide while rules we want to enforce c-) is the actual code following these rules? d-) probably add it to Grunt e-) write about these new rules somewhere

**Comment by denis-sokolov** [comment]
My apologies for not explaining my train of thought. I shall try now.

I do agree with all your points. This would indeed be the way if you were to begin enforcing some rules.
I, however, am absolutely in no position to make such decisions for the project. This pull request is for entirely different, practical reasons.

Namely, in some environments the code is already linted - rules or no rules.
My primary example is my code editor.
It always lints the code I edit.
Even if the project I'm contributing to doesn't enforce any rules, I'm still better off. Without the config file, however, the file is spammed with warnings from the strict default options. By adding the configuration file, I merely document the truth, that some rules in the project are explicitly _not_ followed and make the lint output more useful.

TL;DR: The pull request is for documenting existing practice, not prescribing anything.

**Comment by paglias** [comment]
@denis-sokolov no problem but I'm not going to merge it, while it may be a good discussion to have.

You IDE doesn't allow for the linting rules to be customized?

Otherwise you csn add it to your local repo paying attention not to commit it

**Comment by denis-sokolov** [comment]
> You IDE doesn't allow for the linting rules to be customized?
> Otherwise you csn add it to your local repo paying attention not to commit it

Of course it does, of course I can, that's besides the point.
What about the other new developers – do you want them to need to turn off linting because
you don't have a `.jslintrc`?
Also see CodeClimate. The output is right now way too noisy due to lint warnings that the project
never intended to adhere to. At the same time there are still useful warnings in there.

Feel free to keep it closed, I don't mind at all. :)
Just providing you options and ideas.

**Comment by paglias** [comment]
@denis-sokolov I don't want to make anyone life more difficult, I'm just not so convinced this is the better way to solve the issue even though I understand that this is only relaxing some strict rules set by default.

Anyway let's keep it open and wait for @lefnire to come back from vacation to have him deciding

**Comment by denis-sokolov** [comment]
Today developed on a new machine and hit it again.
Without my pull request, I'd have to come up with all options again.
Luckily, I could grab a file from this pull request.
</shameless promotion>

**Comment by denis-sokolov** [comment]
Another reason to merge this is automatic tools.
CodeClimate show [a million of "issues"](https://codeclimate.com/github/HabitRPG/habitrpg/issues?page=2), most of which are just default jshint warnings, that are not followed in the project.
But among those useless warnings there are a few useful ones that are easy to miss.

@lefnire, please meeeerge.

**Comment by deilann** [comment]
He's on vacay atm, but should be back soon, I think. (I know it's been a
while, but thought you should know.)

**Comment by denis-sokolov** [comment]
Oh, I didn't know that he's still on vacation. Thanks!

**Comment by deilann** [comment]
I think he's back (?) but not back-back? Don't quote me on it.

**Comment by lefnire** [comment]
Just got back today (got to meet @paglias in the flesh!). I see no issues with this PR, if it makes life easier for new devs on certain platforms and seems not to impact users not on those platforms, seems win-win to me? I'll merge it, @paglias if it causes any issues or difficulties for you we can revert - does that sound ok?

**Comment by lefnire** [comment]
May have caused travis errors, see https://travis-ci.org/HabitRPG/habitrpg/builds/26786530 . Might be something else, haven't had a chance to investigate yet

**Comment by denis-sokolov** [comment]
Nah, it's unrelated to this merge:

```
In file included from ../src/CanvasPattern.cc:9:0:
../src/Image.h:20:21: fatal error: gif_lib.h: No such file or directory
compilation terminated.
```

Also, the [pull request build](https://travis-ci.org/HabitRPG/habitrpg/builds/22129144) succeeded back when I proposed it.

Finally, nothing in our build uses jshint anyway.

**Comment by lefnire** [comment]
:+1:
